providers/keycloak.go

```diff
@@ -59,41 +60,39 @@ func NewKeycloakProvider(p *ProviderData) *KeycloakProvider {
 	return &KeycloakProvider{ProviderData: p}
 }
 
-func (p *KeycloakProvider) SetGroup(group string) {
-	p.Group = group
-}
+// EnrichSession uses the Keycloak userinfo endpoint to populate the session's
+// email and groups.
+func (p *KeycloakProvider) EnrichSession(ctx context.Context, s *sessions.SessionState) error {
+	// Fallback to ValidateURL if ProfileURL not set for legacy compatibility
+	userinfoURL := p.ValidateURL.String()
+	if p.ProfileURL != nil && *p.ProfileURL != (url.URL{}) {
```

Comment:
Isn't the point of it being a pointer that it should be non-empty when the pointer is initialised? Is there some special case we are guarding against here?

Reply:
Apparently the `nil` in the defaults for the provider turn into this in the end: https://github.com/oauth2-proxy/oauth2-proxy/blob/d2ffef2c7e0a622295c42cb2fdd70ef14593d03c/providers/provider_data.go#L111

I struggled for a clean way to figure out if the URL truly wasn't set -- I also thought about checking `p.profileURL.String() != ""` maybe that is clearer?

Reply:
We've used `p.profileURL.String() != ""` in various places before to check if it's empty, let's stick with that, it feels clearer to me at least?

Reply:
Perfect - I switched to that pattern to match 👍